### ISSUE TITLE
Demo is broken in some browsers - XKCD's API uses weak encryption, so some browsers reject the API responses

### ISSUE THREAD

**Issue by steenburgh** [steenburgh/React-XKCD-Reader#1]
What the title says: The homepage redirects to /-1 instead of the latest comic on mobile.

_Edit: The title of this issue has changed multiple times as my understanding of it has evolved. Take a look at the comment history or look at the last comment for a quick summary._

**Comment by steenburgh** [comment]
Looks like the app actually just doesn't work at all on mobile. Tested on chrome for Android v52

**Comment by steenburgh** [comment]
Unfortunately, this isn't going to be an easy issue to fix. It looks like XKCD's API doesn't support HTTPS properly, so some browsers will refuse to accept the response with `net::ERR_INSECURE_RESPONSE`. If we change the API url back to the regular HTTP url, we'll instead get a mixed content error, unless we switch over to regular HTTP everything. The problem with doing this is that Firebase forces users to use HTTPS. 

I can think of a couple of options:  
- Move to another hosting platform - something like github pages.
- Write a backend that proxies requests to XKCD's API - Eg. Request from Front End ---> Our Backend ---> XKCD's Endpoint ---> Our Backend ---> Front End - I'm not sure if this is something we can do with Firebase or not.

**Comment by steenburgh** [comment]
I've looked around at other hosting platforms and it's been tough to find one that is both 1. Free and 2. Works well with single page apps. Firebase is simply the best for this. I've spent several hours on multiple occasions trying to fix this. This is a lot of work just for the purpose of just getting a demo up and running, not to mention that this project is meant to be more of a toy project than anything, so I'm calling it quits for now + I'm simply not going to fix this issue.   

The XKCD API should really be using proper HTTPs, and hopefully they'll add it soon so I no longer have to worry about this issue.
